docs/articles/documentation/guides/basic-guides/select-page-elements.md

```diff
@@ -0,0 +1,542 @@
+
+You can also pass DOM element snapshots to test actions.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+const label = Selector('#tried-section').child('label');
+
+test('My Test', async t => {
+    const labelSnapshot = await label();
+
+    await t.click(labelSnapshot);
+});
+```
+
+In this instance, the selector that fetched this snapshot is re-executed before the action.
+
+### Define Assertion Actual Value
+
+You can pass a selector property to an [assertion](assert.md) to check whether the DOM node has the expected state.
+
+```js
+import { Selector } from 'testcafe';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('Assertion with Selector', async t => {
+    const developerNameInput = Selector('#developer-name');
+
+    await t
+        .expect(developerNameInput.value).eql('')
+        .typeText(developerNameInput, 'Peter')
+        .expect(developerNameInput.value).eql('Peter');
+});
+```
+
+When you pass selector properties instead of values, TestCafe enables [Smart Assertion Query Mechanism](assert.md#smart-assertion-query-mechanism) to avoid accidental errors and unstable tests. In this example, the assertion does not fail immediately if the condition does not match. TestCafe recalculates the `developerNameInput.value` property value until the assertion passes or the timeout expires.
+
+## Selector Timeout
+
+When a selector is executed, TestCafe waits for the target node to appear in the DOM until the *selector timeout* expires.
+
+Use the [timeout](../../reference/selector.md#optionstimeout) option to specify the selector timeout in test code. To set the timeout when you launch tests, pass it to the [runner.run](../../reference/testcafe-api/runner.md#run) API method or the [--selector-timeout](../../reference/command-line-interface.md#--selector-timeout-ms) command line option.
+
+During the timeout, the selector is re-executed until it returns a DOM node or the timeout is exceeded. If TestCafe cannot find the corresponding node in the DOM, the test fails.
+
+> Note that you can specify that the node returned by the selector should also be visible. To do this, use the [visibilityCheck](../../reference/selector.md#optionsvisibilitycheck) option.
+
+## Debug Selectors
+
+TestCafe outputs information about failed selectors to test run reports.
+
+When you try to use a selector that does not match any DOM element, the test fails and an error is thrown. The error message indicates which selector has failed.
+
+An error can also occur when you call [selector's methods](#member-tables) in a chain. These methods are applied to the selector one by one. TestCafe detects a method after which the selector no longer matches any DOM element. This method is highlighted in the error message.
```

Comment:
"when you call [selector's methods]" a? the?

Reply:
"TestCafe detects a method after which the selector no longer matches any DOM element" Explain

Reply:
Yes, this sentence is not very clear. How about:

*An error can also occur when you call a [selector's methods](#member-tables) in a chain. These methods are applied to the selector one by one. TestCafe detects the first method that returns no elements and highlights it in the error message.*